src/com/seafile/seadroid2/ui/activity/BrowserActivity.java

```diff
@@ -875,28 +889,49 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
             break;
         case PICK_FILE_REQUEST:
             if (resultCode == RESULT_OK) {
-                if (!Utils.isNetworkOn()) {
-                    ToastUtils.show(this, R.string.network_down);
-                    return;
-                }
 
-                Uri uri = data.getData();
-                String path;
-                try {
-                    path = Utils.getPath(this, uri);
-                } catch (URISyntaxException e) {
-                    e.printStackTrace();
-                    return;
-                }
-                if(path == null) {
-                    ToastUtils.show(this, "Unable to upload, no path available");
-                    Log.i(DEBUG_TAG, "Pick file request did not return a path");
-                    return;
-                }
-                ToastUtils.show(this, getString(R.string.added_to_upload_tasks));
-                //ToastUtils.show(this, getString(R.string.upload) + " " + Utils.fileNameFromPath(path));
-                addUploadTask(navContext.getRepoID(),
-                    navContext.getRepoName(), navContext.getDirPath(), path);
+                Log.i(DEBUG_TAG, "Got uri: "+data.getData());
+
+                new AsyncTask<Uri, Void, File>() {
+
+                    @Override
+                    protected File doInBackground(Uri... params) {
+                        try {
+                            Uri uri = params[0];
+                            File tempDir = new File(DataManager.getExternalTempDirectory(), "upload-"+System.currentTimeMillis());
```

Comment:
@forouher but after assembled the displayName and `temDir` can avoid name conflicts, is it?